src/distributors/local-installer.ts

```diff
@@ -0,0 +1,60 @@
+import * as tc from '@actions/tool-cache';
+
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+import { macOSJavaContentDir, setupFromJdkFile } from '../util';
+
+import { JavaBase } from './base-installer';
+import {
+  JavaInstallerOptions,
+  JavaDownloadRelease,
+  JavaInstallerResults
+} from './base-models';
+
+export class LocalDistributor extends JavaBase {
+  private jdkFile: string;
+  constructor(installerOptions: JavaInstallerOptions) {
+    super('local_distributor', installerOptions);
+    this.jdkFile = installerOptions.jdkFile!;
```

Comment:
I think we shouldn't pass `jdkFile` as a part of JavaInstallerOptions. Probably, we should pass it as a separate input parameter

Reply:
Done. Thank you.